### ISSUE TITLE
page-break-lines should be an optional dependency

### ISSUE THREAD

**Issue by Luis-Henriquez-Perez** [emacs-dashboard/emacs-dashboard#160]
I don't really understand why this package is a _required_ dependency. It seems like it should be quite possible to display the content of the dashboard without page breaks. Or even, if you insist on some delineation you could have this:  `---------------------------------` .

**Comment by JesusMtnez** [comment]
I agree that `page-break-lines` could be made an optional dependency. Right now I do not have plenty of time to work on something like that. But feel free work on that and open a PR. I'll try to give feedback whenever I can :smile:

**Comment by Luis-Henriquez-Perez** [comment]
Ok sounds good. 👍 I was already planning to PR some things once I get more familiar with the code.

**Comment by Luis-Henriquez-Perez** [comment]
Hey, so I'm more familiar with the code now. Looks like it's going to be really easy to remove `page-break-lines` as a dependency. It's only used 2 times in the code.

The integration between dashboard and `page-break-lines` is so thin that it seems that the ideal solution, would be to just not have page-break-lines in dashboard at all and have the user do this if they want the integration.

```elisp
(leaf page-break-lines
  :hook (dashboard-mode-hook . page-break-lines-mode))

(leaf dashboard
  :ensure t
  :require t
  :setp
  (dashboard-page-separator . "\n\f\n")
  (dashboard-center-content . t))
```
The benefit of doing this is that it's natural and simple. In general I think packages should not talk about one another unless one package is fundamental to the use of the other. If you choose not to go about it this way, then we'd need to add some sort of clunky option like `dashboard-use-page-break-lines` and add more if statements to the code.

On a related note, when I was reading `dashboard-insert-startupify-lists` I noticed the code below. 

```elisp
(when dashboard-center-content
            (goto-char (car (last dashboard--section-starts)))
            (let ((margin (floor (/ (max (- (window-width) max-line-length) 0)  2))))
              (while (not (eobp))
                (and (not (eq ? (char-after)))
                     (insert (make-string margin ?\ )))
                (forward-line 1))))
```
The point of it seems to be not to insert the margin before a page break. Right now it will only work if the page break is the default one `^L`, but if the user provides a custom page break it will insert the margin and the user's page-break will be really off (this happened to me). I think this code should be generalized (perhaps using `string-match-p` or something similar to check whether the text matches `dashboard-page-separator`).

**Comment by JesusMtnez** [comment]
I would like to see this PR to drop the requirement of `page-break-lines`. As far as I understood, it makes sens to remove it. Thank you so much for digging into this one.

BTW, I see you use `leaf.el`. Any reason in particular to move to `leaf.el` from `use-package`? I've seen it got a lot of traction recently, and I've been thinking of moving to `leaf.el` since it has `straight.el` integration in `leaf-keywords.el`.

**Comment by Luis-Henriquez-Perez** [comment]
The main reason I made the change that it's _much_ easier to extend and to understand it because it's lower level. I don't have to know about things like `use-package-process-keywords` or `use-package-concat` to extend leaf. Customizing it is essentially just pushing forms into lists. The only values you need to know are `leaf-name` (the name of the package) and `leaf-body` the value of the keyword you're extending.

I highly recommend taking a look at [the notes](https://github.com/Luis-Henriquez-Perez/.emacs.d/blob/no-use-package/main.org#leaf) I took on leaf if you're interested in seeing how to customizing it. Had I read these, I would have been customizing it like a pro in about 10 minutes.

I'll create the pull request momentarily.

**Comment by JesusMtnez** [comment]
Thank you so much for the reference, I'll go through it for sure :smile: 

I'll try to review as soon as I can, I've been kind of busy lately :worried:

**Comment by Luis-Henriquez-Perez** [comment]
Ok. I hope it's helpful. :smile: 

No rush, I just removed two lines in the PR.

**Comment by simpers** [comment]
Any news on this? When I first start the dashboard buffer the lines aren't showing (I prefer this) but they show on a redraw and will then stay for the rest of the time. Which is a shame, since they change the whole aesthetics of the dashboard :thinking:

**Comment by jcs090218** [comment]
With my configuration [jcs-emacs](https://github.com/jcs090218/jcs-emacs), it displays sequence of characters when inside terminal. I think I am able to manage it so it display even with inside the graphical user interface.

I think this is indeed a good enhancement!
